core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/verticalscale/CoreVerticalScaleService.java

```diff
@@ -84,4 +133,122 @@ public void updateTemplateWithVerticalScaleInformation(Long stackId, StackVertic
+            StackDto stackDto, String service) {
+        Map<String, String> config = new HashMap<>();
+        StringBuilder localMountPaths = new StringBuilder("");
+        StringBuilder logMountPaths = new StringBuilder("");
+        if (awsInstanceStorageInfo != null && !awsInstanceStorageInfo.isEmpty()) {
+            int instanceStorageCount = awsInstanceStorageInfo.get(0).getInstanceStorageCount();
+            int count = 1;
+            while (instanceStorageCount > 0) {
+                if ("YARN".equalsIgnoreCase(service)) {
+                    localMountPaths.append("/hadoopfs/ephfs").append(count).append("/nodemanager");
+                    logMountPaths.append("/hadoopfs/ephfs").append(count).append("/nodemanager/log");
+                } else if ("IMPALA".equalsIgnoreCase(service)) {
+                    localMountPaths.append("/hadoopfs/ephfs").append(count).append("/impala/scratch");
+                    logMountPaths.append("/hadoopfs/ephfs").append(count).append("/impala/datacache");
+                }
+                if (instanceStorageCount - 1 > 0) {
+                    localMountPaths.append(',');
+                    logMountPaths.append(',');
+                }
+                count++;
+                instanceStorageCount--;
+            }
+        }
+        if ("YARN".equalsIgnoreCase(service)) {
+            config.put(YARN_LOCAL_DIR, localMountPaths.toString());
+            config.put(YARN_LOG_DIR, logMountPaths.toString());
+        } else if ("IMPALA".equalsIgnoreCase(service)) {
+            config.put(IMPALA_SCRATCH_DIR, localMountPaths.toString());
+            config.put(IMPALA_DATACACHE_DIR, logMountPaths.toString());
+        }
+        return config;
+    }
+
+    private void setTemporaryStorageOnTemplate(Template template, List<AwsInstanceStorageInfo> instanceStoreInfo) {
+        if (instanceStoreInfo != null && !instanceStoreInfo.isEmpty()) {
+            template.setTemporaryStorage(TemporaryStorage.EPHEMERAL_VOLUMES);
+            template.setInstanceStorageCount(instanceStoreInfo.get(0).getInstanceStorageCount());
+            template.setInstanceStorageSize(instanceStoreInfo.get(0).getInstanceStorageSize());
+        } else {
+            template.setTemporaryStorage(TemporaryStorage.ATTACHED_VOLUMES);
+            template.setInstanceStorageCount(0);
+            template.setInstanceStorageSize(0);
+        }
+    }
+
+    private void pollClouderaManagerServices(ClusterApi clusterApi, String service, String status) throws Exception {
+        int readCount = 1;
+        while (readCount < MAX_READ_COUNT) {
+            Map<String, String> readResults = clusterApi.clusterModificationService().readServices();
+            if (status.equals(readResults.get(service.toLowerCase()))) {
+                return;
+            }
+            sleep(SLEEP_INTERVAL);
+            readCount++;
+        }
+    }
```

Comment:
i tried using the standard poller - but as soon as the command is executed, it throws a COMMAND EXITED error everytime for just stopping a specific service. That is the reason i created this method

Reply:
Please do not introduce new polling engine, CB has already more approaches than it should

Reply:
@lajosrodek @doktoric This is what I tried and it threw a "Command Exited exception"
I first created a ApiCommand for starting/stopping a specific service. Get the ApiCommand's BigDecimal ID and use 
`ExtendedPollingResult startDefaultPolling(StackDtoDelegate stack, ApiClient apiClient, BigDecimal commandId, String commandName) {` to do polling.

Reply:
Sorry, I still do not see how `ApiCommand` comes into the picture. The condition you were using here comprises the following 2 statements:
```
            Map<String, String> readResults = clusterApi.clusterModificationService().readServices();
            if (status.equals(readResults.get(service.toLowerCase()))) {
                return;
            }
```
This would be the exit condition of the `com.dyngr.Polling` instance if you were to follow the approach of `ExternalDatabaseService.startDatabase()`.